src/pip/_internal/utils/glibc.py

```diff
@@ -21,7 +18,7 @@ def glibc_version_string_confstr() -> Optional[str]:
         return None
     try:
         # os.confstr("CS_GNU_LIBC_VERSION") returns a string like "glibc 2.17":
-        _, version = os.confstr("CS_GNU_LIBC_VERSION").split()
+        _, version = os.confstr("CS_GNU_LIBC_VERSION").split()  # type: ignore[union-attr]
```

Comment:
It's ignoring the case where it returns None, because we already catch an AttributeError one line below

From https://docs.python.org/3/library/os.html#os.confstr

> If the configuration value specified by name isn’t defined, None is returned

Reply:
I can change so we don't catch AttributeError though!

Reply:
Theoratically Windows shouldn’t reach this line at all since there’s a check above (which is why Mypy does not complain). But it’s probably still best to keep the AttributeError just in case.